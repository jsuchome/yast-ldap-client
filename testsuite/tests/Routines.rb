# encoding: utf-8

#  File:	Routines.ycp
#  Summary:	Test of routines.ycp functions
#  Author:	Jiri Suchomel <jsuchome@suse.cz>
#  $Id$
module Yast
  class RoutinesClient < Client
    def main
      # testedfiles: Ldap.ycp
      Yast.import "Testsuite"

      @READ = { "target" => { "tmpdir" => "/tmp", "stat" => {} } }
      Testsuite.Init([@READ], 0)

      Yast.import "Ldap"

      Yast.include self, "ldap/routines.rb"

      @dn = "uid=root,dc=suse,dc=cz"

      Testsuite.Test(lambda { Ldap.get_rdn(@dn) }, [], 0)
      Testsuite.Test(lambda { Ldap.get_cn(@dn) }, [], 0)
      Testsuite.Test(lambda { Ldap.get_new_dn("admin", @dn) }, [], 0)

      @dn = "uid=root"

      Testsuite.Test(lambda { Ldap.get_rdn(@dn) }, [], 0)
      Testsuite.Test(lambda { Ldap.get_cn(@dn) }, [], 0)
      Testsuite.Test(lambda { Ldap.get_new_dn("admin", @dn) }, [], 0)

      @dn = "root"

      Testsuite.Test(lambda { Ldap.get_rdn(@dn) }, [], 0)
      Testsuite.Test(lambda { Ldap.get_cn(@dn) }, [], 0)
      Testsuite.Test(lambda { Ldap.get_new_dn("admin", @dn) }, [], 0)

      nil
    end
  end
end

Yast::RoutinesClient.new.main
