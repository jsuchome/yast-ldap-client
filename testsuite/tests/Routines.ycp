/**
 *  File:	Routines.ycp
 *  Summary:	Test of routines.ycp functions
 *  Author:	Jiri Suchomel <jsuchome@suse.cz>
 *  $Id$
 */
{
    // testedfiles: Ldap.ycp
    import "Testsuite";

    import "Ldap";

    string dn	= "uid=root,dc=suse,dc=cz";

    Testsuite::Test (``(Ldap::get_rdn (dn)), [], 0);
    Testsuite::Test (``(Ldap::get_cn (dn)), [], 0);
    Testsuite::Test (``(Ldap::get_new_dn ("admin", dn)), [], 0);

    dn	= "uid=root";

    Testsuite::Test (``(Ldap::get_rdn (dn)), [], 0);
    Testsuite::Test (``(Ldap::get_cn (dn)), [], 0);
    Testsuite::Test (``(Ldap::get_new_dn ("admin", dn)), [], 0);

    dn	= "root";

    Testsuite::Test (``(Ldap::get_rdn (dn)), [], 0);
    Testsuite::Test (``(Ldap::get_cn (dn)), [], 0);
    Testsuite::Test (``(Ldap::get_new_dn ("admin", dn)), [], 0);
}
