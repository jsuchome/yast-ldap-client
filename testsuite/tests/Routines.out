Return	uid=root
Return	root
Return	cn=admin,dc=suse,dc=cz
Return	uid=root
Return	root
Return	cn=admin
Return	root
Return	root
Return	cn=admin
